src/main/java/fruitshop/service/reader/FileReaderImpl.java

```
@@ -0,0 +1,21 @@
+package fruitshop.service.reader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String fileName) throws IOException {
```

This violates checklist item: 'Don't ignore exceptions in try-catch blocks'. The method throws IOException directly, but the requirements specify that IOExceptions should be caught and rethrown as RuntimeExceptions with informative messages and the original cause.